### PR TITLE
Initial project scaffolding

### PR DESCRIPTION
**Pull request by adrianchiris**
- Add github issue templates
- Add github workflows (build, test, lint, image push, codeql)
- Setup golangci-lint
- Setup mockery
- Implement stub application with basic functionality
- Add Dockerfile
- Add README and CONTRIBUTING doc
- Add various automations using Task

### CODE REVIEW

File: cmd/nic-feature-discovery/app/options/options.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	}
+
+	if o.FeatureFileName == "" {
+		return fmt.Errorf("feature file name cannot be empty")
+	}
+
+	return err
+}
+
+func folderExist(path string) error {
```
Comment (moshe010) [inline cmd/nic-feature-discovery/app/options/options.go:77]: can you move folderExist to utils?

**Review by moshe010 — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -7,15 +7,31 @@
...
 
 # Go workspace file
 go.work
+
+# editor and IDE
+.idea
+.vscode
+*.swp
+*.swo
+*~
```
Comment (ykulazhenkov) [inline .gitignore:37]: nit: no new line at the end

File: .golangci.yaml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  funlen:
+    lines: 120
+    statements: 55
+  goconst:
+    min-len: 2
+    min-occurrences: 2
+  gocyclo:
+    min-complexity: 30
+  goimports:
+    local-prefixes: gitlab-master.nvidia.com/cloud-orchestration/k8s-net-dra-driver
```
Comment (ykulazhenkov) [inline .golangci.yaml:112]: invalid prefix

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# Build with make to apply all build logic defined in Makefile
+RUN task build
+
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static-debian11:latest
+WORKDIR /
+COPY --from=builder /workspace/build/nic-feature-discovery .
+
+
```
Comment (ykulazhenkov) [inline Dockerfile:25]: nit: extra new line at the end

File: taskfiles/Version.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  GIT_TAG_LAST: 
+    sh: git describe --tags --abbrev=0 2>/dev/null || true
+  VERSION: "{{.VERSION | default .GIT_TAG_LAST}}"
+  RELEASE_STATUS:
+    sh: echo '{{if or (ne .GIT_TAG "") (ne .VERSION .GIT_TAG_LAST)}},released{{end}}'
+  VERSION_LDFLAGS: -X github.com/Mellanox/nic-feature-discovery/pkg/utils/version.version={{.VERSION}}
+    -X github.com/Mellanox/nic-feature-discovery/pkg/utils/version.commit={{.COMMIT}}
+    -X github.com/Mellanox/nic-feature-discovery/pkg/utils/version.gitTreeState={{.GIT_TREE_STATE}}
+    -X github.com/Mellanox/nic-feature-discovery/pkg/utils/version.releaseStatus={{.RELEASE_STATUS}}
+    -X github.com/Mellanox/nic-feature-discovery/pkg/utils/version.date={{.DATE}}
```
Comment (ykulazhenkov) [inline taskfiles/Version.yaml:25]: no new line at the end

File: cmd/nic-feature-discovery/app/app.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+NVIDIA CORPORATION, its affiliates and licensors retain all intellectual
+property and proprietary rights in and to this material, related
+documentation and any modifications thereto. Any use, reproduction,
+disclosure or distribution of this material and related documentation
+without an express license agreement from NVIDIA CORPORATION or
+its affiliates is strictly prohibited.
+
+SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES
+SPDX-License-Identifier: LicenseRef-NvidiaProprietary
```
Comment (ykulazhenkov) [inline cmd/nic-feature-discovery/app/app.go:12]: Should we use this license for the project?

File: cmd/nic-feature-discovery/main.go [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+package main
+
+import (
+	"github.com/Mellanox/nic-feature-discovery/cmd/nic-feature-discovery/app"
+	"k8s.io/component-base/cli"
+	"k8s.io/klog/v2"
+)
+
+func main() {
+	klog.FlushAndExit(klog.ExitFlushTimeout, cli.Run(app.NewNICFeatureDiscoveryCommand()))
```
Comment (ykulazhenkov) [inline cmd/nic-feature-discovery/main.go:23]: `cli.Run` will flush logs, I think `klog.FlushAndExit` is not requried

File: .golangci.yaml (hunk 3 above)
Comment (adrianchiris) [reply .golangci.yaml:112]: changed to correct prefix

File: .gitignore (hunk 2 above)
Comment (adrianchiris) [reply .gitignore:37]: fixed

File: Dockerfile (hunk 4 above)
Comment (adrianchiris) [reply Dockerfile:25]: fixed

File: taskfiles/Version.yaml (hunk 5 above)
Comment (adrianchiris) [reply taskfiles/Version.yaml:25]: fixed

File: cmd/nic-feature-discovery/app/app.go (hunk 6 above)
Comment (adrianchiris) [reply cmd/nic-feature-discovery/app/app.go:12]: good catch ! this one needs to be apache 2.0 leme fix

File: cmd/nic-feature-discovery/main.go (hunk 7 above)
Comment (adrianchiris) [reply cmd/nic-feature-discovery/main.go:23]: looking into Run implementation, yes i see it flushes them. 
will remove this one then.

**Review by ykulazhenkov — APPROVED** [review]

**Pull request merged**